### PR TITLE
Rspec test for is_obstructed. Factorybot attempt - commented out-

### PR DESCRIPTION
**Pull request by guillaumebihet**
Rspec tests implemented for is_obstructed? method.

Attempt to use FactoryBot to build the game unsuccessful so far. Either creating validation issues or not properly evaluating.  I left the code (commented out) of some of my attempts in the first is_obstructed? rspec test and in factories.rb for consultation.

### CODE REVIEW

**Comment by KyleDuca** [comment]
Lets hold off from merging this. Lets see if we can figure out the cause and if we have to make some changes first.

**Review by KyleDuca — DISMISSED** [review]
Lets wait and figure out the problem first. I'm having the same problem and there's no point if we have try to fix it later. Sorry, thanks.

**Comment by ilyakrasnov** [comment]
My guess is that factorybot fails because of non existent users so far.. For the sake of is_obstructed method, if the test are passing, we can merge it now and refactor to use factory bot at a later stage (probably next week as soon as you'll need to write more tests).

File: app/models/piece.rb [hunk 1]
```diff
@@ -1,6 +1,79 @@
 class Piece < ApplicationRecord
   belongs_to :game
 
+  def is_obstructed?(game, pos2)
+    
+    @occupied_positions =[]
```
Comment (ilyakrasnov) [inline app/models/piece.rb]: no need for '@'. its used to declare instance variables... in this case you just use it as a local variable inside your method

File: app/models/piece.rb [hunk 2]
```diff
@@ -1,6 +1,79 @@
...
+  end
+
+  def check_squares
+    (@squares_to_check & @occupied_positions).length > 0? true : false  
+  end
+
+  def is_vertically_obstructed?(pos1,pos2)
+    if @x1 == @x2
+      @y1 , @y2 = @y2 , @y1 if @y1 > @y2
+      for y in (@y1+1...@y2)
```
Comment (ilyakrasnov) [inline app/models/piece.rb]: Maybe we can use a more 'rubyish' construct, `each`..

```
(@y1+1...@y2).each do |y|
...
end
```
An interesting discussion on differences: https://jacopretorius.net/2012/02/for-vs-each-in-ruby.html

File: app/models/piece.rb [hunk 3]
```diff
@@ -1,6 +1,79 @@
...
+    end
+      
+    
+
+    @occupied_positions 
+    # defining the variables for the methods
+    
+    pos1=[self.position_x,self.position_y]
+
+    @x1=pos1[0]
```
Comment (ilyakrasnov) [inline app/models/piece.rb]: same with '@' here

File: spec/controllers/games_controller_spec.rb [hunk 4]
```diff
@@ -7,4 +7,66 @@
...
+    it "should return true because it is obstructed" do
+      
+
+      # game = FactoryBot.create(:game)
+      # game = FactoryBot.build(:game, white_player_id: 1, black_player_id: 2)
+      
+      game = Game.new
+      piece1 = Piece.create(position_x: 7, position_y: 7, game_id: game.id)
+      piece2 = Piece.create(position_x: 7, position_y: 5, game_id: game.id)
+      game.pieces << piece1 << piece2
```
Comment (ilyakrasnov) [inline spec/controllers/games_controller_spec.rb]: as you wrote on slack, we can save this line by doing

```
piece1 = game.pieces.create(...)
```

**Comment by ilyakrasnov** [comment]
Nice work, looks good 👍

**Review by ilyakrasnov — APPROVED** [review]
Code seems solid 👍

**Pull request merged**